### ISSUE TITLE
In Outlook Preview mailbox.item.body.setSelectedDataAsync removes prefix spaces

### ISSUE THREAD

**Issue by koller-patrick** [OfficeDev/office-js#3496]
> Message from office-js bot:
We’re closing this issue because it has been inactive for a long time. We’re doing this to keep the issues list manageable and useful for everyone. If this issue is still relevant for you, please create a new issue.
> Thank you for your understanding and continued feedback.

# In the new Outlook Preview mailbox.item.body.setSelectedDataAsync removes prefix spaces

## Your Environment

* Platform: PC desktop Windows 11
* Host: Outlook Preview
* Office version number: 1.2023.616.100
* Operating System: Windows
 
## Expected behavior

When inserting text containing a space as prefix (" test") in an open mail item, I would expect that the space prefix is inserted too in the message and not only the text ("test"). As it is working in the "normal", not Preview Outlok version.

## Current behavior

When inserting text containing a space as prefix (" test") in an open mail item, the space prefix is removed and does not appear in the mail message text.

## Steps to reproduce

1. Insert text with the following code multiple times:
```
const insertTextWithSpacePrefix = async () => {
        Office.context.mailbox.item.body.setSelectedDataAsync(
            " test with prefix space",
            {
                coercionType: Office.CoercionType.Text,
            },
            async (asyncResult: Office.AsyncResult<void>) => {
                console.log(asyncResult);
            }
        );
    };
```
2. Space prefix of the text is removed:
![image](https://github.com/OfficeDev/office-js/assets/71555268/4aa45e58-3691-46e5-850d-f971266789db)

## Context
As the space prefix is not removed in the "normal" Outlook and this behavior only occurs in the new preview version, I would expect the same behavior in both Outlook versions.

Thank you for taking the time to report an issue. Our triage team will respond to you in less than 72 hours. Normally, response time is <10 hours Monday through Friday. We do not triage on weekends.

**Comment by exextoc** [comment]
Thanks for reporting this issue regarding setSelectedDataAsync. It has been put on our backlog. We unfortunately have no timelines to share at this point.

Internal tracking id: Office: [[216672]

**Comment by mobisw-msft** [comment]
Hey @koller-patrick ,

This issue has ben fixed and is available from build 20240529002 onwards. Do reopen this issue if this issue still persists.
